src/pkg/boltz/boltzService.ts

```diff
@@ -0,0 +1,372 @@
+import type { CreateSwapCommonResponse, NetworkString, SubmarineSwapResponse } from './boltz';
+import { Boltz } from './boltz';
+import { randomBytes } from 'crypto';
+import type { BIP174SigningData, OwnedInput } from 'liquidjs-lib';
+import {
+  address,
+  AssetHash,
+  Blinder,
+  Creator,
+  CreatorInput,
+  crypto,
+  Extractor,
+  Finalizer,
+  networks,
+  Pset,
+  script as bscript,
+  script,
+  Signer,
+  Transaction,
+  Updater,
+  witnessStackToScriptWitness,
+  ZKPGenerator,
+  ZKPValidator,
+} from 'liquidjs-lib';
+import type { TagData } from 'bolt11';
+import bolt11 from 'bolt11';
+import type { Unspent } from '../../domain/chainsource';
+import type { ECPairInterface } from 'ecpair';
+import zkp from '@vulpemventures/secp256k1-zkp';
+import { fromSatoshi } from '../../extension/utility';
+
+export interface BoltzServiceInterface {
+  getBoltzPair(pair: string): Promise<any>;
+  createSubmarineSwap(
+    invoice: string,
+    network: NetworkString,
+    refundPublicKey: string
+  ): Promise<SubmarineSwap>;
+  createReverseSubmarineSwap(
+    claimPublicKey: Buffer,
+    network: NetworkString,
+    invoiceAmount: number
+  ): Promise<ReverseSwap>;
+  makeClaimTransaction(p: MakeClaimTransactionParams): Promise<Transaction>;
+  getInvoiceExpireDate(invoice: string): number;
+  getInvoiceValue(invoice: string): number;
+}
+
+export interface SubmarineSwap {
+  address: string;
+  expectedAmount: number;
+  id: string;
+  redeemScript: string;
+}
+
+export interface ReverseSwap {
+  blindingPrivateKey: string;
+  claimPublicKey: string;
+  id: string;
+  invoice: string;
+  lockupAddress: string;
+  preimage: Buffer;
+  redeemScript: string;
+  timeoutBlockHeight: number;
+}
+
+interface MakeClaimTransactionParams {
+  utxo: Unspent;
+  claimPublicKey: Buffer;
+  claimKeyPair: ECPairInterface;
+  preimage: Buffer;
+  redeemScript: Buffer;
+  destinationScript: Buffer;
+  fee: number;
+  password: string;
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export class BoltzService implements BoltzServiceInterface {
+  private boltz: Boltz;
+  private network: NetworkString;
+
+  constructor(network: NetworkString) {
```

Comment:
constructor should be agnostic in my opinion
```suggestion
  constructor(url: string) {
```